core/src/main/java/io/questdb/griffin/engine/table/FilterOnValuesRecordCursorFactory.java

```diff
@@ -81,20 +82,29 @@ public FilterOnValuesRecordCursorFactory(
                 addSymbolKey(SymbolTable.VALUE_NOT_FOUND, symbol, indexDirection);
             }
         }
-        if (orderByMnemonic == OrderByMnemonic.ORDER_BY_INVARIANT) {
+        if (orderByMnemonic == OrderByMnemonic.ORDER_BY_INVARIANT && !orderByTimestamp) {
             rowCursorFactory = new SequentialRowCursorFactory(cursorFactories, cursorFactoriesIdx);
         } else {
             rowCursorFactory = new HeapRowCursorFactory(cursorFactories, cursorFactoriesIdx);
```

Comment:
What if it the order is by timestamp descending here, will HeapRowCursorFactory do descending order?

Reply:
Conditionals in SqlCodeGenerator should prevent that from happening . HeapRowCursorFactory doesn't support descending order . I can add it if needed .